### PR TITLE
[RUMV3] allow metricsets inside transactions

### PR DESCRIPTION
**Pull request by jalvz**
## Motivation/summary
RUM V3 spec will allow metricsets inside transactions, in addition to top level events. 

Either case, the RUM agent will not report transaction fields. When inside the transaction (breakdown metrics), the apm-server will derive and fill the fields from the transaction.

## Checklist

- [x] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)

## How to test these changes

`make test`

### CODE REVIEW

File: testdata/intake-v3/rum_events.ndjson [hunk 1]
```diff
@@ -1,6 +1,3 @@
...
+{"me": {"y": {"t": "Processing"},"sa": {"ysc": {"v": 1},"yss": {"v": 124}}}}
```
Comment (simitt) [inline testdata/intake-v3/rum_events.ndjson:3]: Is that a valid case that a metricset belongs to a span but not to a transaction?

Comment (jalvz) [reply testdata/intake-v3/rum_events.ndjson:3]: Yeah, probably doesn't make much sense, but I wanted to keep a top level event to test they are ingested. RUM doesn't send system metrics yet (they will soon) so there aren't  "real" examples we can use.

File: docs/data/intake-api/generated/rum_v3_events.ndjson [hunk 2]
```diff
@@ -1,6 +1,3 @@
...
+{"me": {"y": {"t": "Processing"},"sa": {"ysc": {"v": 1},"yss": {"v": 124}}}}
```
Comment (vigneshshanmugam) [inline docs/data/intake-api/generated/rum_v3_events.ndjson:3]: is this a valid metricset? We can send metricset inside transaction or even in the same events as of transaction?

Comment (vigneshshanmugam) [reply docs/data/intake-api/generated/rum_v3_events.ndjson:3]: Asking because how do you differentiate them from custom metricsets like CPU, Memory etc.

Comment (jalvz) [reply docs/data/intake-api/generated/rum_v3_events.ndjson:3]: https://github.com/elastic/apm-server/pull/3733/#discussion_r419396153

:see_no_evil:

Comment (jalvz) [reply docs/data/intake-api/generated/rum_v3_events.ndjson:3]: I don't think it is needed to differentiate, we just accept both, but if it comes inside the transactions then we fill in the transaction info

Comment (jalvz) [reply docs/data/intake-api/generated/rum_v3_events.ndjson:3]: maybe you (and @simitt ) were thinking about having different, separated specs for system and breakdowns, but I think that would just complicate things (more types internally, etc).

Comment (vigneshshanmugam) [reply docs/data/intake-api/generated/rum_v3_events.ndjson:3]: Cool, thanks for the details Juan.

**Pull request merged**